diesel/src/types/mod.rs

```diff
@@ -249,10 +249,20 @@ pub type VarChar = Text;
 #[cfg(feature = "postgres")]
 pub use pg::types::sql_types::*;
 
+pub trait ProvidesSqlTypeFor<DB> where DB: TypeMetadata {
```

Comment:
I would love to see a doc comment here telling me what this type is all about. I assume it abstracts over structs and enums providing SQL types? Maybe also mention `HasSqlType` is implemented for everything that implements this trait.